plugins/modules/route53_health_check.py

```diff
@@ -489,17 +563,52 @@ def main():
     if existing_check:
         check_id = existing_check.get('Id')
 
+    # If update or delete Health Check based on ID
+    update_delete_by_id = False
+    if module.params.get('health_check_id'):
```

Comment:
Rather than having two variables (`id_to_update_delete` and `check_id`) I think we should update the code at line 561:564 to be: 
```
existing_check = find_health_check(ip_addr_in, fqdn_in, type_in, request_interval_in, port_in)

if module.params.get('health_check_id'):
    # Prefer the ID specified by the user
    check_id = module.params.get('health_check_id')
elif existing_check:
    check_id = existing_check.get('Id') 
```
It has the same effect of acting on an ID provided by the user regardless of what ID was returned by `find_health_check()`, but with fewer conditionals later in the code (like the block currently at lines 574:579 under `if state_in == 'absent').

We might want to add a check though to give a warning if the ID returned by `find_health_check()` is different from `module.params.health_check_id`, since having data in `existing_check` for the wrong ID could affect changed status.  @markuman What do you think?

Reply:
> We might want to add a check though to give a warning if the ID returned by find_health_check() is different from module.params.health_check_id, since having data in existing_check for the wrong ID could affect changed status. @markuman What do you think?

If `module.params.health_check_id` is provided, `find_health_check()` should/must not be invoked imo.

>> [1. introduce a new id parameter that updates existing health checks](https://github.com/ansible-collections/community.aws/pull/1143#issuecomment-1128009773)  
>> ...
>> [If the id is not provided, keep the behaviour as it is, just throw warnings that it will change in the future...](https://github.com/ansible-collections/community.aws/pull/1143#issuecomment-1128051370)

But it is invoked, and overwritten later in L571, if `health_check_id` is provided. FMPOV the logic looks correct. There _might_ be room to improve the condition tree.  

The existing integration test passes and also the appended that treated the `tag:Name` is unique key also.  
What currently is missing are some tests that uses the `health_check_id` to update and delete health checks. But I guess @mandar242 is still in progress with it.